adsrefpipe/models.py

```diff
@@ -0,0 +1,256 @@
+        self.parser = parser
+
+    def toJSON(self):
+        """
+        :return: values formatted as python dict, if no values found returns empty structure, not None
+        """
+        return {
+            'bibcode': self.bibcode,
+            'source_filename': self.source_filename,
+            'resolved_filename': self.resolved_filename,
+            'parser': self.parser,
+        }
+
+
+class History(Base):
+    __tablename__ = 'history'
+    __table_args__ = (ForeignKeyConstraint( ['bibcode', 'source_filename'], ['reference.bibcode', 'reference.source_filename']),)
+    id = Column(Integer, primary_key=True)
+    bibcode = Column(String)
+    source_filename = Column(String)
+    source_modified = Column(DateTime)
+    status = Column(String, ForeignKey('action.status'))
+    date = Column(DateTime, default=func.now())
+    total_ref = Column(Integer)
+
+    def __init__(self, bibcode, source_filename, source_modified, status, date, total_ref):
+        """
+
+        :param bibcode:
+        :param source_filename:
+        :param source_modified:
+        :param status:
+        :param date:
+        :param total_ref:
+        """
+        self.bibcode = bibcode
+        self.source_filename = source_filename
+        self.source_modified = source_modified
+        self.status = status
+        self.date = date
+        self.total_ref = total_ref
+
+    def toJSON(self):
+        """
+        :return: values formatted as python dict, if no values found returns empty structure, not None
+        """
+        return {
+            'bibcode': self.bibcode,
+            'source_filename': self.source_filename,
+            'source_modified': self.source_modified,
+            'status': self.status,
+            'date': self.date,
+            'total_ref' : self.total_ref,
+        }
+
+
+class Resolved(Base):
+    __tablename__ = 'resolved'
+    history_id = Column(Integer, ForeignKey('history.id'), primary_key=True)
+    item_num = Column(Integer, primary_key=True)
+    reference_str = Column(String, primary_key=True)
+    bibcode = Column(String)
+    score = Column(Numeric)
+
+    def __init__(self, history_id, item_num, reference_str, bibcode, score):
+        """
+
+        :param history_id:
+        :param item_num
+        :param reference_str:
+        :param bibcode:
+        :param score:
+        """
+        self.history_id = history_id
+        self.item_num = item_num
+        self.reference_str = reference_str
+        self.bibcode = bibcode
+        self.score = score
+
+    def toJSON(self):
+        """
+        :return: values formatted as python dict, if no values found returns empty structure, not None
+        """
+        return {
+            'history_id': self.history_id,
+            'reference_str': self.reference_str,
+            'bibcode': self.bibcode,
+            'score': self.score,
+            'item_num': self.item_num
+        }
+
+
+class Compare(Base):
```

Comment:
`ClassicComparison` or some such thing; `Compare` is a verb (shouldn't be a verb), and it is too vague

Reply:
This should actually go away once classic is turned of. what it holds right now is how classic resolved the record in comparison to reference service. I shall put comment in there what it is for to be clear.